### PR TITLE
return net.DNSError{}

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	if err != nil {
+		// TODO: extract information from err.(type)
+		var timeout bool
+		var temporary bool
+		var notfound bool
+
+		return &net.DNSError{
+			Err:         err.Error(),
+			Server:      server,
+			Name:        name,
```
Comment (karasz) [inline tools.go:19]: if you pass a `*dns.Msg` why you need a `name`? What is wrong with `*dns.Msg.Question[0].Name`?

File: tools.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+func validateResp(server, name string, r *dns.Msg, err error) error {
+	if err != nil {
+		// TODO: extract information from err.(type)
+		var timeout bool
+		var temporary bool
+		var notfound bool
+
+		return &net.DNSError{
+			Err:         err.Error(),
+			Server:      server,
```
Comment (karasz) [inline tools.go:18]: I have kind of mixed feelings about `Server` mostly on n the plus side though

File: tools.go (hunk 1 above)
Comment (amery) [reply tools.go:19]: Nothing wrong, just ignorance

**Review by karasz — COMMENTED** [review]
See comments

File: tools.go (hunk 1 above)
Comment (karasz) [reply tools.go:19]: btw, don't be mislead by the `0` in `Question[0]`. Even if the standard permits multiple `Question` bundle, afaik there is no DNS sever implementation that supports that, so `Question[0]` is pretty much the only `Question`

Comment (amery) [reply tools.go:21]: Amended

**Comment by amery** [comment]
@karasz amended

**Review by karasz — APPROVED** [review]
LGTM, although I am not convinced about the `validateResp()`.

**Comment by amery** [comment]
`v0.1.2` tagged

**Pull request merged**